app/src/main/java/org/mozilla/fenix/settings/advanced/LocaleAdapter.kt

```diff
@@ -120,25 +121,36 @@ class LocaleViewHolder(
 
 class SystemLocaleViewHolder(
     view: View,
-    private val interactor: LocaleSettingsViewInteractor,
-    private val selectedLocale: Locale
-) : BaseLocaleViewHolder(view) {
+    selectedLocale: Locale,
+    private val interactor: LocaleSettingsViewInteractor
+) : BaseLocaleViewHolder(view, selectedLocale) {
     private val icon = view.locale_selected_icon
     private val title = view.locale_title_text
     private val subtitle = view.locale_subtitle_text
 
     override fun bind(locale: Locale) {
         title.text = itemView.context.getString(R.string.default_locale_text)
         subtitle.visibility = View.GONE
-        icon.isVisible = locale === selectedLocale
-
+        icon.isVisible = isCurrentLocaleSelected(locale, isDefault = true)
         itemView.setOnClickListener {
             interactor.onDefaultLocaleSelected()
         }
     }
 }
 
-abstract class BaseLocaleViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+abstract class BaseLocaleViewHolder(
+    view: View,
+    private val selectedLocale: Locale
+) : RecyclerView.ViewHolder(view) {
+
+    fun isCurrentLocaleSelected(locale: Locale, isDefault: Boolean): Boolean {
```

Comment:
nit: `protected`

Reply:
Changed to internal since we need it for testing